render.c

```diff
@@ -33,7 +33,23 @@ static void set_color_for_state(cairo_t *cairo, struct swaylock_state *state,
 	}
 }
 
-void render_frame_background(struct swaylock_surface *surface) {
+static void surface_frame_handle_done(void *data, struct wl_callback *callback,
+		uint32_t time) {
+	struct swaylock_surface *surface = data;
+
+	wl_callback_destroy(callback);
+	surface->frame = NULL;
+
+	render(surface);
```

Comment:
Why are we rendering straight away after we handle the frame done? Wouldn't we render a frame, commit it, wait for the done event, then just immediately render again?

Reply:
`render` checks at the beginning if the surface is dirty and bails if that is not the case, stopping the loop.

My original intent was to move all the commits *out* of render, but render (before: render_frame_background) makes a temporary buffer that it destroys, so commit has to happen within that function. Going the other way and having it absorb the callback was a quicker solution.

Reply:
I see. The names are a bit confusing being `render` and `render_frame` but that can be cleaned up later.